formats/hocon/src/main/kotlin/kotlinx/serialization/hocon/serializers/ConfigMemorySizeSerializer.kt

```diff
@@ -0,0 +1,70 @@
+package kotlinx.serialization.hocon.serializers
+
+import com.typesafe.config.*
+import java.math.BigInteger
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.hocon.*
+
+/**
+ * Serializer for [ConfigMemorySize].
+ * For decode using [HOCON Size in bytes format](https://github.com/lightbend/config/blob/main/HOCON.md#size-in-bytes-format).
+ * For encode used format for powers of two: byte, KiB, MiB, GiB, TiB, PiB, EiB, ZiB, YiB.
+ * Encoding use the largest value of format.
+ * Example:
+ *  1024 byte -> 1 KiB;
+ *  1024 KiB -> 1 MiB;
+ *  1025 KiB -> 1025 KiB.
+ * Usage example:
+ * ```
+ * @Serializable
+ * data class ConfigMemory(
+ *      @Serializable(ConfigMemorySizeSerializer::class)
+ *      val size: ConfigMemorySize
+ * )
+ * val config = ConfigFactory.parseString("size = 1 MiB")
+ * val configMemory = Hocon.decodeFromConfig(ConfigMemory.serializer(), config)
+ * val newConfig = Hocon.encodeToConfig(ConfigMemory.serializer(), configMemory)
+ * ```
+ */
+@ExperimentalSerializationApi
+object ConfigMemorySizeSerializer : KSerializer<ConfigMemorySize> {
+
+    // For powers of two.
+    private val memoryUnitFormats = listOf("byte", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB")
+
+    private val valueResolver: (Config, String) -> ConfigMemorySize = { conf, path -> conf.decodeMemorySize(path) }
+
+    override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("hocon.com.typesafe.config.ConfigMemorySize", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): ConfigMemorySize = if (decoder is HoconDecoder) {
```

Comment:
Here and in other serializers (and documentation samples for HoconEncoder/Decoder):

IMO, this formatting looks hard to follow because:
1. This is not a one-line `if`, but it starts from the same line as declaration
2. `if` has braces, while `else` does not (even though there's a single statement under `if`).

I suggest writing it from the new line and being consistent with an absence of braces:

```
override fun deserialize(decoder: Decoder): ConfigMemorySize =
    if (decoder is HoconDecoder)
        decoder.decodeValue(valueResolver)
    else
        throw ...
```
(to save some linebreaks, it is possible to collapse `if (decoder is HoconDecoder) decoder.decodeValue(valueResolver)` into one line and `else throw` in another)

If you find braces preferable for some reason, put them both in `if` and `else`.

Reply:
Fixed